### PR TITLE
trace/perf: Add support for collecting metrics with perf

### PR DESCRIPTION
**Pull request by marcbonnici**

### CODE REVIEW

File: devlib/trace/perf.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+                                       self.target.abi, 'perf')
+        return self.target.install(host_executable)
+
+    def _build_commands(self):
+        commands = []
+        for opts, label in zip(self.optionstrings, self.labels):
+            commands.append(self._build_perf_command(opts, self.events, label))
+        return commands
+
+
```
Comment (setrofim) [inline devlib/trace/perf.py]: Extra blank line.

File: devlib/trace/perf.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        perf list
+
+    Whereas options, they can be provided as a single string as following ::
+
+        perf_options = '-a -i'
+
+    Options can be obtained by running the following in the command line ::
+
+        man perf-record
```
Comment (setrofim) [inline devlib/trace/perf.py]: "man perf-stat"

File: devlib/trace/perf.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        if isinstance(optionstring, list):
+            self.optionstrings = optionstring
+        else:
+            self.optionstrings = [optionstring]
+        if self.events and isinstance(self.events, basestring):
+            self.events = [self.events]
+        if not self.labels:
+            self.labels = ['perf_{}'.format(i) for i in range(len(self.optionstrings))]
+        if len(self.labels) != len(self.optionstrings):
+            raise HostError('The number of labels must match the number of optstrings provided for perf.')
```
Comment (setrofim) [inline devlib/trace/perf.py]: `ValueError` -- `HostError` implies a problem on the host, not with user-specified arguments.

**Comment by marcbonnici** [comment]
Updated

**Pull request merged**